site-src/geps/gep-1282.md

```diff
@@ -98,12 +106,297 @@ Of course, from the section above, you can see that there are many other feature
+    // characteristics in common: they are tightly bound to the backend, 
+    // rather than being important at the route level, and the service owner
+    // (application developer in the Gateway API personas) should control 
+    // these settings, not the owner of the Gateway implementation (the 
+    // Cluster Operator in the Gateway API personas).
+    // 
+    // Support: Extended
+    // 
+    // +required
+    Spec BackendCapabilitiesSpec `json:”spec”`
+
+    // Status represents the current state of the backend and its capabilities.
+    // +optional
+    Status *BackendCapabilitiesStatus `json:”status,omitempty”`
+}
+
+// BackendTargetReference identifies an API object to apply BackendCapabilities to. 
+// 
+type BackendTargetReference struct {
+    // Group is the group of the target resource.
+    Group Group `json:"group"`
+
+    // Kind is kind of the target resource.
+    // Valid values include:
+    // * "Service" (the default if no Kind is specified)
+    // * "ServiceImport"
+    //
+    // Invalid values include:
+    //
+    // * "invalid/kind" - "/" is an invalid character
+    // +required
+    Kind string `json:"kind"`
+
+    // Name is the name of the target resource.
+    Name ObjectName `json:"name"`
+}
+
+// BackendCapabilitiesSpec defines a group of specific properties held by its
+// affiliated backend, and specified by the backend owner (typically the 
+// application developer).  
+//
+// Examples of properties that belong in the BackendCapabilitiesSpec include:
+// TLS re-encryption, session state, timeouts, retries, Websockets, HTTP/2, 
+// load balancer algorithm, healthcheck info, external traffic policy,
+// connection limits, error page, protocol switch, alerts, and others. 
+//
+// Support: Extended
+//
+type BackendCapabilitiesSpec struct {
+    // Backend refers to the object to which we apply these capabilities
+	// +required
+	Backend backendTargetReference `json:"backend,omitempty"`
+	
+    // TLS is the section of details on encryption processing in the traffic 
+    // hop from Gateway to backend, sometimes known as upstream TLS or re-encrypt.
+    // A xRoute configured as passthrough cannot specify a Forward encrypt type.
+    // TLS is useful for HTTPRoute and should not be used on UDP/TCP/TLSRoute.
+    //
+    // +optional
+    TLS *EncryptType `json:"tls,omitempty"`
+}
+
+// EncryptType describes the details needed to implement encryption from the
+// gateway to the backend.
+type EncryptType struct {
+    // The minimal TLSVersion that is accepted by the backend.
+    // If empty, there is no minimal backend TLSVersion.
+    // +optional	
+    MinVersion   *TLSVersion `json:"minVersion,omitempty"`
+
+    // The maximum TLSVersion that is accepted by the backend.
+    // If empty, there is no maximum backend TLSVersion.
+    // +optional    
+    MaxVersion   *TLSVersion  `json:"maxVersion,omitempty"`
+    
+    // CipherSuite is the minimum cipher suite that the backend accepts.
```

Comment:
There is no "minimum" cipher suite as far as I can tell; this should be a list?

Reply:
Agreed, this should be `[]string` so that it can store a list of ciphers in the usual format.

Reply:
TLS 1.3 only accepts a subset of what 1.1/1.2 accept.  I changed this to a list so the implementation can validate whatever cipher suites are included.  However, if they were left empty, and there is no min or maxVersion, the implementation would need to determine which cipherSuites to use.

Reply:
TLSv1.2 cipher suites and TLSv1.3 cipher suites don't map one-to-one.  One option would be to have two separate fields for TLSv1.2 (and earlier) and for TLSv1.3.  That's what HAProxy does: 
- http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#3.1-ssl-default-bind-ciphers
- http://cbonte.github.io/haproxy-dconv/2.6/configuration.html#3.1-ssl-default-bind-ciphersuites